tools/jit/gen_jit_dispatch.py

```diff
@@ -274,16 +270,27 @@ def declkey(decl):
                 ])
                 decl['has_tensor_options'] = True
 
-    jit_decls = sort_decls(jit_decls)
-    for decl in jit_decls:
-        ops.append(OPERATOR.substitute(signature=signature(decl),
-                                       op=emit_decl_variant(decl)))
-
-    # Sort the generated snippets to ensure that the generation is deterministic
-    env = {
-        'constructors': ops,
-    }
-    write(out, 'register_aten_ops.cpp', REGISTER_ATEN_OPS_CPP, env)
+    # Group and sort the generated snippets to ensure that the
+    # generation is deterministic
+    jit_decl_groups = sort_decls(jit_decls)
+
+    # NOTE: see the comment at the top of the register_aten_ops.cpp
+    # template regarding sharding of the generated files.
+    num_shards = 3
+    shards = [[] for _ in range(num_shards)]
+
+    # ops are assigned arbitrarily but stably to a file based on hash
+    for group in jit_decl_groups:
+        x = sum(ord(c) for c in group[0]['name']) % num_shards
```

Comment:
@zdevito, I know you suggested this sharding strategy, but wouldn't it be better to use an actual hash function here? Like, it's not like Python makes it hard to do so. (I agree that this particular use site doesn't matter how we do it; it's more of a "is this code going to get copy pasted" concern)

Reply:
well this already is a copy-paste of the other site :)

I could switch to just calling hash(). It looks like it's deterministic across interpreter startups (I wasn't sure about this, but I've just tested it). It may vary between python versions, or maybe even between machines or environments? Not sure about this, or if it matters even if true

Reply:
No, `hash` is not guaranteed to be consistent, and in fact won't be in many situations. We need to have our own custom function